functions-continued/how-to-read-documentation.md

```diff
@@ -0,0 +1,171 @@
+
+- Recognize the version of the docs
+- Recognize the sections of the docs
+- Read through the function signature
+- Take time to understand and rework the examples
+
+## Recognize Versions and Sections
+
+As we know, software changes over time. Take one piece of software, and the codebase for it today, compared to the codebase it had two years ago. They may have the same name, but different functions, implementations, details, and results! A set of docs is often tied to a specific version of that API.
+
+The version of the docs you use should **match** the version of the library or API you're using.
+
+### !callout-info
+
+## Example: Comparing Python 2 and Python 3 docs
+
+Let's compare the docs between [Python 2](https://docs.python.org/2/library/functions.html) and [Python 3](https://docs.python.org/3/library/functions.html). The docs are visually incredibly different. We can see that there are plenty of differences between the docs, even though they're both Python. Python 2 has a `cmp()` function, where Python 3 does not. Python 3 has a `bytes()` function, and Python 2 doesn't. The consequences of using mismatched versions of docs and tools could result in a lot of broken code!
+
+### !end-callout
+
+Documentation can be huge in size. [Python 3's documentation](https://docs.python.org/3/) has a lot of material to cover, and a lot of sections!
+
+Commonly, docs reserve a sidebar or column to for navigating different methods and different pages. Use these to help you navigate!
+
+Sections and words to look out for:
+
+- The word **"reference"**
+- The words **"methods"** or "**functions"**
+- Lists of function names
+- Sections dedicated to the _kind_ of functionality you're looking for by it's purpose
+- A search bar
+- For external APIs, the word "endpoints"
+
+## Look for the Function Signature
+
+Once we find a function we want to study, we should study the function signature defined in the docs.
+
+We should always look for these things:
+
+1. An explanation of the function's responsibility
+2. What the function returns
+3. An explanation of the parameters
+4. Special cases or notes
+
+**Look at this example:** Let's look at the docs for [Python's `sum()` function](https://docs.python.org/3/library/functions.html#sum)
+
+<!-- Written in 2020 with Python 3 docs -->
+
+> `sum(iterable, /, start=0)`
+>
+> Sums _start_ and the items of an _iterable_ from left to right and returns the total. The _iterable_'s items are normally numbers, and the start value is not allowed to be a string.
+>
+> ...
+
+From this example:
+
+1. This method is responsible for summing an _iterable_ and a _start_, usually numbers.
+2. The function returns the total sum
+3. This function always takes in the positional argument `iterable`. It has a _keyword argument_ `start`, which has a default value of `0`.
+
+### !callout-info
+
+## What's with the `/`?
+
+Python docs have a unique pattern: the docs will put a `/` inbetween the positional and keyword arguments
```

Comment:
Are students familiar with the differences between these types of arguments? If so, I think this is good as is. But if not, it would be helpful to add some context

Reply:
Great point! There needs to be the small lesson written, but the plan is for Keyword Args to be introduced before this. :)